operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -1676,8 +1709,60 @@ def getSSLOrNone(server):
   except:
     trace("Ignoring getSSL() exception, this is expected.")
     ret = None
+
   return ret
 
+def getRealSSLListenPort(server, sslport):
+  """
+  Return the real ssl listening port that will be used in runtime.
+   This is the actual port that WebLogic will bind to.  This occurs when user
+   specify 7002 in the model or wlst offline when creating the domain which results empty
+   entry in the config.xml. The introspector using wlst offline to read the domain and the
+   mbean returns 8100. we cannot use this in the topology when setting up the container port
+   since the actual listening port is 7002.
+
+   If it is not a server template, then just return from the mbean.
+
+  :param server:  server or server template
+  :param sslport: sslport from wlst offline mbean
+  :return: listening port
+  """
+  if server_template_sslports.has_key(server.getName()):
+    configxml_ssl_port = server_template_sslports[server.getName()]
+    if configxml_ssl_port is None:
```

Comment:
what if configxml_ssl_port is not None but has the value of 7002?

Reply:
Then configxml_ssl_port.getListenPort (actually it's a template) returns 7002,  the server_template_listening_ports contains the actual value in the config.xml  or None if no entry.  Since we only care the case where there is no entry,  I let the case to turn from the mbean.

Reply:
In this case, since the server here is a template, WLS will treat the value as 7002, but mbean will return 8100.

Reply:
Looks like you are correct, even if 7002 is actually in the config.xml, it still returns 8100. I will change it

Reply:
fixed

Reply:
Somehow this use case still does not work.

Reply:
It's a string comparison issue, fixed.